src/range.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2018 King's College London
+// created by the Software Development Team <http://soft-dev.org/>
```

Comment:
This whole file is verbatim modulo renamings?

Reply:
Not quite, but not far off either. And I wasn't really sure what license to use. In retrospect, I might as well just use the Rust license in here.